inference-engine/src/preprocessing/ie_preprocess_gapi.cpp

```diff
@@ -768,53 +854,140 @@ void PreprocEngine::checkApplicabilityGAPI(const Blob::Ptr &src, const Blob::Ptr
         IE_THROW()  << "Unsupported input blob type: expected MemoryBlob, NV12Blob or I420Blob";
     }
 
-    // dst is always a memory blob
-    if (!dst->is<MemoryBlob>()) {
-        IE_THROW()  << "Unsupported network's input blob type: expected MemoryBlob";
-    }
-
     const auto &src_dims = src->getTensorDesc().getDims();
-    const auto &dst_dims = dst->getTensorDesc().getDims();
 
     // dimensions sizes must be equal if both blobs are memory blobs
-    if (!yuv420_blob && src_dims.size() != dst_dims.size()) {
+    if (!yuv420_blob && src_dims.size() != network_input_valid_dims.size()) {
         IE_THROW() << "Preprocessing is not applicable. Source and destination blobs "
                               "have different number of dimensions.";
     }
-    if (dst_dims.size() != 4) {
-        IE_THROW() << "Preprocessing is not applicable. Only 4D tensors are supported.";
-    }
 
     // dimensions must not have values that are equal to 0
     if (has_zeros(src_dims)) {
         IE_THROW() << "Invalid input data dimensions: " << details::dumpVec(src_dims);
     }
+}
+
+void PreprocEngine::checkBatchedBlobApplicabilityGAPI(const BatchedBlob::Ptr &src, const SizeVector &network_input_valid_dims) {
+    size_t inner_blob_count = src->size();
+    size_t index = 0;
+    try {
+        for ( ; index < inner_blob_count; index++) {
+            checkSingleBlobApplicabilityGAPI(src->getBlob(index), network_input_valid_dims);
+        }
+    }
+    catch (const std::exception& e) {
+        IE_THROW() << "Fail in BatchedBlob validation: " << e.what() << " - by index: " << index
+                   << ", total count: "  << inner_blob_count;
+    }
```

Comment:
Not sure if this extra exception handling is required, I'd replace it with proper DEBUG-side logging

Reply:
for my sadness - IE has no any Logger  implementation. Exception is only way to communicate with outer world

Reply:
yeah and _that_ exception is thrown already - no need to catch & rethrow - that's my point

Reply:
it adds extended information for error description: what bb, what index inside bb contains invalid data. from my perspective if user spent not 3 days for debugging IE preprocessing code but 3 hours for adjustment misprinted self-written-code - it always better

Please consider simple example with comparison of catching in user-application 
- just  `std::bad_alloc with -1` 
and 
- `cannot create mu_super_table by index: 6, error - std::bad_alloc`